[kab-dot] iP

kab-dot opened the pull request:
Aether

> “By failing to prepare, you are preparing to fail.” – Benjamin Franklin [(source)](https://www.projectmanager.com/blog/planning-quotes)

Aether helps you prepare for the future, so success is guaranteed! It's

- text-based
- easy to learn
- ~~FAST~~ SUPER FAST to use

Here's what you need to do!

1. download it from [here](https://nus-cs2103-ay2324s2.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉 AND prepare for the days ahead
And it is _FREE!_

Features:

 

- [ ] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)
If you are a **Java** programmer, you can use it to practice Java too. Here's the main method:

```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

khushi833 committed:
rename, greet, exit

khushi833 committed:
rename, greet, exit

khushi833 committed:
echo + study + bye

khushi833 committed:
add, list

khushi833 committed:
mark and unmark

khushi833 committed:
todos, events, deadlines

khushi833 committed:
text-ui-testing

khushi833 committed:
handle errors

khushi833 committed:
delete

khushi833 committed:
date and time

khushi833 committed:
OOP refactoring

khushi833 committed:
packages + fixed file save issues

khushi833 committed:
Merge commit 'a6f73244a20539e20c2189973ebf597df34ef631'

* commit 'a6f73244a20539e20c2189973ebf597df34ef631':
  Bump gradle and lib version
  Add Gradle support
  docs/README.md: Tweak document template

khushi833 committed:
gradle

khushi833 committed:
Implement the JUnit tests for TaskList and Event as well as Gradle (implemented in the previous commit)

khushi833 committed:
Wrote JavaDocs for most of the public methods and classes

khushi833 committed:
Implement search and find

khushi833 committed:
Implement Coding Standard

khushi833 committed:
Merge branch 'branch-A-CodingStandard'
Fix the merge conflict
* branch-A-CodingStandard:
  Implement Coding Standard

# Conflicts:
#	src/main/java/duke/Command.java

khushi833 committed:
Merge branch 'branch-A-JavaDoc'
Fix merge conflicts
* branch-A-JavaDoc:
  Wrote JavaDocs for most of the public methods and classes

# Conflicts:
#	src/main/java/duke/Command.java
#	src/main/java/duke/DukeException.java
#	src/main/java/duke/Storage.java
#	src/main/java/duke/Task.java
#	src/main/java/duke/Ui.java

khushi833 committed:
Test out the code after merging

PateShin commented on src/main/java/duke/Command.java:
```diff
@@ -0,0 +1,124 @@
+    public enum CommandType {
+        BYE, LIST, TODO, DEADLINE, EVENT, DELETE, MARK, UNMARK, FIND, INVALID
+
+    }
```
There are some issues with line break including CommandType enum. Consider to remove them 😄

PateShin commented on src/main/java/duke/Command.java:
```diff
@@ -0,0 +1,124 @@
+     * Enumerates the possible types of commands.
+     */
+    private CommandType type;
+    private String argument;
```
Consider changing the name ```argument``` to something more descriptive, such as **```taskDescription```**.

kab-dot replied:
Yes, will do!

PateShin commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,97 @@
+            throw new DukeException("I'm sorry, but I don't know what that means :-(");
+        }
+    }
+
+    private Command parseFindCommand(String input) throws DukeException {
+        String keyword = input.substring(4).trim();
+        if (keyword.isEmpty()) {
+            throw new DukeException("OOPS!!! The keyword for find cannot be empty.");
+        }
+        return new Command(Command.CommandType.FIND, keyword);
+    }
+}
```
I noticed that Javadocs are missing in some places, including in ```parseFindCommand```, ```findTasksByKeyword```, and ```showMatchingTasks```.

Consider writing Javadoc as an example:
```
// Example of Javadoc

/**
 * This method calculates the sum of two integers.
 *
 * @param a The first integer.
 * @param b The second integer.
 * @return The sum of the two integers. 
 */
public int calculateSum(int a, int b) {
    return a + b;
}
```

kab-dot replied:
Okay, yup will take note!

PateShin reviewed (commented):
Overall, the functionality and naming are mostly clear to read👍 

Consider to remove any dead codes in **Event**, **Deadline**, **Duke** and **Parser**.
There are also some minor issues, such as line breaks and comments. I recommend checking the line breaks in every file.

PateShin commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,105 @@
+//}
+package duke;
+
+import java.time.DateTimeException;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
```
I found that these are not in alphabetical order. Consider reordering them.

PateShin reviewed (commented):
This is just for add on. I found that these are not in alphabetical order in **Deadline** Class. Consider reordering them.

```
import java.time.DateTimeException;
import java.time.LocalDateTime;
import java.time.format.DateTimeFormatter;
```

khushi833 committed:
Fix Checkstyle Errors

khushi833 committed:
Fix some minor comments that should be removed

khushi833 committed:
Merge branch 'A-CheckStyle'

* A-CheckStyle:
  Fix some minor comments that should be removed

khushi833 committed:
Implement GUI

khushi833 committed:
Merge branch 'Level-10'

* Level-10:
  Implement GUI

khushi833 committed:
Add pictures

khushi833 committed:
Fix the package name in FXML

khushi833 committed:
Implement GUI

khushi833 committed:
Fix the GUI and commit

VatsalVitosh commented on src/main/java/duke/Command.java:
```diff
@@ -0,0 +1,244 @@
+
+        StringBuilder response = new StringBuilder();
+        switch (type) {
+        case BYE:
```
The execute method could be more concise. Perhaps you could break down this method into smaller methods, each responsible for a specific command type.

Example:

```ruby
public String execute(TaskList tasks, Ui ui, Storage storage) throws DukeException {
    switch (type) {
        case BYE:
            return executeBye();
        case LIST:
            return executeList(tasks);
        case TODO:
            return executeTodo(tasks, storage);
        // ...
        default:
            throw new DukeException("Unexpected command type: " + type);
    }
}

private String executeBye() {
    return "Goodbye! Hope to see you again soon!";
}

private String executeList(TaskList tasks) {
    // ...
}

private String executeTodo(TaskList tasks, Storage storage) throws DukeException {
    // ...
}
```

kab-dot replied:
Oh, okay will definitely take that into consideration.

VatsalVitosh commented on src/main/java/duke/Event.java:
```diff
@@ -0,0 +1,68 @@
+            this.startTime = LocalDateTime.parse(startTime, formatter);
+            this.endTime = LocalDateTime.parse(endTime, formatter);
+        } catch (DateTimeParseException e) {
+            System.out.println("Invalid date/time format. Please use: yyyy-MM-dd HH:mm");
```
Consider using a constant for the date format and declaring it as private static final field at the top of the class. Doing so would make the method more generalised and if you needed to change the format, you would only need to change the constant instead of going to each method where it's used and changing it.

Example:
```ruby
private static final String DATE_TIME_FORMAT = "yyyy-MM-dd HH:mm";
```

Similarly for the declarations in Deadline.java

VatsalVitosh commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,60 @@
+package duke;
+/**
```
Consider adding line separation between package declaration and JavaDoc

VatsalVitosh commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,62 @@
+    public Storage(String filePath) {
+        this.filePath = filePath;
+    }
+    /**
```
Perhaps review all of the code to add line separation between the code itself and the JavaDoc headers.

VatsalVitosh commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,238 @@
+    }
+
+    public void showSeparator() {
+        addMessage("_____________________________");
```
Consider using a constant for the separator line to ensure consistency across different methods.

Example:
```ruby
private static final String SEPARATOR_LINE = "_____________________________";
```

Doing so, you won't have to copy/paste the line over and over again and can just
```ruby
addMessage(SEPARATOR_LINE);
```

kab-dot replied:
yup!

VatsalVitosh reviewed (commented):
The code, overall, is well constructed and closely follows the coding standard. I've made a few suggestions to potentially make the code a little more concise :)

khushi833 committed:
Fix the gradle run issue and other checkstyle issues

khushi833 committed:
Add Assertions

kab-dot committed:
Merge pull request #1 from kab-dot/branch-A-Assertions

Add Assertions

khushi833 committed:
Refactor code for improved code quality and conciseness
Improved readability and maintainability by extracting repetitive code into helper methods.
Reduced redundancy and improved efficiency by optimizing the execute method in the command class, resulting in shorter and more concise code.

kab-dot committed:
Merge pull request #2 from kab-dot/branch-A-CodeQuality

Refactor code for improved code quality and conciseness

khushi833 committed:
Implement duplicate task detection
- Added functionality to detect and handle duplicate tasks when adding tasks to the task list. A new method `containsTask` was added to the `TaskList` class to check for duplicates based on task description.

Add graceful shutdown process for 'bye' command
- Implemented a feature where the application displays a farewell message and waits for two seconds before closing after the user inputs the 'bye' command.
- This provides a smoother user experience by allowing users to see the farewell message and understand that the application is intentionally closing.

khushi833 committed:
Merge branch 'C-DetectDuplicates'

* C-DetectDuplicates:
  Implement duplicate task detection - Added functionality to detect and handle duplicate tasks when adding tasks to the task list. A new method `containsTask` was added to the `TaskList` class to check for duplicates based on task description.

khushi833 committed:
Update the User Guide and fix all other exceptions
Remove commented code and add javadocs

khushi833 committed:
Add the product screenshot

khushi833 committed:
Edit the User Guide

khushi833 committed:
Fix User Guide
